### PR TITLE
Resteasy: do not use Jandex's ClassInfo in a Set

### PR DESCRIPTION
**Pull request by aloubyansky**
since it implements neither equals() nor hashCode()

### CODE REVIEW

File: extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java [hunk 1]
```diff
@@ -229,19 +229,21 @@ public void build(
...
-                    if (!additionalPaths.contains(annotation)) { // scanned resources only contains real JAX-RS resources
-                        scannedResources.putIfAbsent(clazz.name(), clazz);
-                    }
-                    reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, className));
+                    if (!withoutDefaultCtor.containsKey(clazz.name())) {
+                        String className = clazz.name().toString();
+                        if (!additionalPaths.contains(annotation)) { // scanned resources only contains real JAX-RS resources
+                            scannedResources.putIfAbsent(clazz.name(), clazz);
+                        }
+                        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, className));
```
Comment (geoand) [inline extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:242]: Shouldn't these be outside of the `if (!withoutDefaultCtor.containsKey(clazz.name()))` check?

Comment (aloubyansky) [reply extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:242]: If it in `withoutDefaultCtor` it's already been processed, hasn't it?

Comment (geoand) [reply extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:242]: That is certainly true, but the point of `withoutDefaultCtor` wasn't really to guard against that case.
I do understand what your checking here, it just seems odd to reuse `withoutDefaultCtor` to perform this check.

Comment (aloubyansky) [reply extensions/resteasy-server-common/deployment/src/main/java/io/quarkus/resteasy/server/common/deployment/ResteasyServerCommonProcessor.java:242]: I could introduce another Set<DotName> for that purpose.

**Comment by geoand** [comment]
Let's get this one in for now

**Review by geoand — APPROVED** [review]

**Pull request merged**